### PR TITLE
[storage] fix a issue in getting latest snapshot version

### PR DESCRIPTION
**Pull request by areshand**
### Description
The problem is the txn version can be smaller than snapshot version during restore. 

The existing code doesn't correctly get the snapshot version for restore case and cause the wrong snapshot used

### Test Plan

manual test on previous failed cases and see it succeeded

### CODE REVIEW

File: storage/backup/backup-cli/src/coordinators/replay_verify.rs [hunk 1]
```diff
@@ -177,11 +178,13 @@ impl ReplayVerifyCoordinator {
...
 
-        let txn_manifests = transactions.into_iter().map(|b| b.manifest).collect();
         TransactionRestoreBatchController::new(
             global_opt,
             self.storage,
-            txn_manifests,
+            transactions
+                .into_iter()
+                .map(|t| t.manifest)
+                .collect::<Vec<_>>(),
```
Comment (msmouse) [inline storage/backup/backup-cli/src/coordinators/replay_verify.rs:187]: is it necessary?

File: testsuite/replay_verify.py [hunk 2]
```diff
@@ -202,7 +202,7 @@ def main(runner_no=None, runner_cnt=None, start_version=None, end_version=None):
         runner_end = end_version
 
     # run replay-verify in parallel
-    N = 16
+    N = 1
```
Comment (msmouse) [inline testsuite/replay_verify.py:205]: recover?

**Review by msmouse — APPROVED** [review]
Update the summary please.
I assume it's manually but thoroughly tested? (smoke/unitest test that can reproduce the previous failure would be nice)

**Review by grao1991 — APPROVED** [review]

File: storage/backup/backup-cli/src/coordinators/replay_verify.rs (hunk 1 above)
Comment (areshand) [reply storage/backup/backup-cli/src/coordinators/replay_verify.rs:187]: not necessary. found them equivalent so didn't change it back

**Review by msmouse — APPROVED** [review]

**Pull request merged**